src/operations/connect.ts

```diff
@@ -561,6 +552,32 @@ function translateOptions(options: any) {
   // Set the socket and connection timeouts
   if (options.socketTimeoutMS == null) options.socketTimeoutMS = 360000;
   if (options.connectTimeoutMS == null) options.connectTimeoutMS = 10000;
+
+  const translations: any = {
+    // SSL translation options
+    sslCA: 'ca',
+    sslCRL: 'crl',
+    sslValidate: 'rejectUnauthorized',
+    sslKey: 'key',
+    sslCert: 'cert',
+    sslPass: 'passphrase',
+    // SocketTimeout translation options
+    socketTimeoutMS: 'socketTimeout',
+    connectTimeoutMS: 'connectionTimeout',
+    // Replicaset options
+    replicaSet: 'setName',
+    rs_name: 'setName',
+    secondaryAcceptableLatencyMS: 'acceptableLatency',
+    connectWithNoPrimary: 'secondaryOnlyConnectionAllowed',
+    // Mongos options
+    acceptableLatencyMS: 'localThresholdMS'
+  };
+
+  for (const name in options) {
+    if (translations[name]) {
+      options[translations[name]] = options[name];
+    }
```

Comment:
Is it intentional to leave the original option behind? 
options.ca and options.sslCA will coexist for example

Reply:
yes, this is a raw port of the existing `translateOptions` method. I agree its cleaner to pluck options and create a new set of options, but we'll be doing that as part of the mongo client options work.